### ISSUE TITLE
Exception: Git clone failed

### ISSUE THREAD

**Issue by ahmed-moha** [leoafarias/fvm#725]
fvm install http
Can't load Kernel binary: Invalid kernel binary format version.
Creating local mirror...
 Counting objects:    [██████████████████████████████████████████████████] 100%
 Compressing objects: [██████████████████████████████████████████████████] 100%
 Receiving objects:   [██................................................] 4%
✓ Clone complete

Exception: Git clone failed

Please run command with  --verbose if you want more information

**Comment by leoafarias** [comment]
@ahmed-moha I need more info then this, as described in the template you posted.  Running the --verbose flag will provide more insight on the error

**Comment by zainulhassan815** [comment]
I'm also having similar issue:

```
Argument information:
  Top level options:
  - verbose: true

Command: install

Creating local mirror...
 Counting objects:    [██████████████████████████████████████████████████] 100%
 Receiving objects:   [██................................................] 4%
✓ Clone complete

Exception: Git clone failed

package:fvm/src/utils/parsers/git_clone_update_printer.dart 92    runGitCloneUpdate
package:fvm/src/services/flutter_service.dart 146                 FlutterService.updateLocalMirror
package:fvm/src/services/flutter_service.dart 30                  FlutterService._ensureCacheDir
package:fvm/src/services/flutter_service.dart 195                 FlutterService.getReference
package:fvm/src/services/flutter_service.dart 166                 FlutterService.isTag
package:fvm/src/workflows/ensure_cache.workflow.dart 193          validateFlutterVersion
package:fvm/src/workflows/ensure_cache.workflow.dart 25           ensureCacheWorkflow
package:fvm/src/commands/install_command.dart 75                  InstallCommand.run
package:args/command_runner.dart 212                              CommandRunner.runCommand
package:fvm/src/runner.dart 246                                   FvmCommandRunner.runCommand
package:fvm/src/runner.dart 122                                   FvmCommandRunner.run
package:scope/src/scope.dart 151                                  Scope.run
/private/tmp/fvm-20240524-7481-6aoj9j/fvm-3.1.4/bin/main.dart 13  main
```

**Comment by zhouzhuo810** [comment]
I'm also having similar issue：

```
Creating local mirror...

✓ Clone complete

Exception: Git clone failed

package:fvm/src/utils/parsers/git_clone_update_printer.dart 92  runGitCloneUpdate
package:fvm/src/services/flutter_service.dart 146               FlutterService.updateLocalMirror
package:fvm/src/services/flutter_service.dart 30                FlutterService._ensureCacheDir
package:fvm/src/services/flutter_service.dart 195               FlutterService.getReference
package:fvm/src/services/flutter_service.dart 166               FlutterService.isTag
package:fvm/src/workflows/ensure_cache.workflow.dart 193        validateFlutterVersion
package:fvm/src/workflows/ensure_cache.workflow.dart 25         ensureCacheWorkflow
package:fvm/src/commands/use_command.dart 131                   UseCommand.run
package:args/command_runner.dart 212                            CommandRunner.runCommand
package:fvm/src/runner.dart 246                                 FvmCommandRunner.runCommand
package:fvm/src/runner.dart 122                                 FvmCommandRunner.run
package:scope/src/scope.dart 151                                Scope.run
/Users/runner/work/fvm/fvm/bin/main.dart 13                     main

```

**Comment by roundspecs** [comment]
I am also having a similar issue:
```
$ fvm install 3.16.3
Creating local mirror...
 Counting objects:    [██████████████████████████████████████████████████] 100%
 Compressing objects: [██████████████████████████████████████████████████] 100%
 Receiving objects:   [██................................................] 4%
✓ Clone complete

Exception: Git clone failed

Please run command with  --verbose if you want more information
```

**Comment by leoafarias** [comment]
I have improved the logging on version 3.1.5, hopefully we can get a bit more insight on what the error is.

**Comment by rsilvr** [comment]
@leoafarias Might be just a coincidence but I'm experiencing a similar issue with [puro](https://github.com/pingbird/puro)...
Anyway, below is my output in version 3.1.5 in a clean install:
```
fvm global stable --verbose                                                                                                                        

Argument information:
  Top level options:
  - verbose: true

Command: global

Flutter SDK: Channel: Stable is not installed.
✔ Would you like to install it now? · yes                                                                                                                                                     
Creating local mirror...
 Counting objects:    [██████████████████████████████████████████████████] 100%
 Compressing objects: [██████████████████████████████████████████████████] 100%
 Receiving objects:   [██................................................] 4%Cloning into '/home/silvr/fvm/cache.git'...

remote: Enumerating objects: 514205, done.        
remote: Counting objects:  89% (183/205)        
remote: Counting objects: 100% (205/205), done.        
remote: Compressing objects:  18% (26/141)        
remote: Compressing objects:  41% (58/141)        
remote: Compressing objects:  97% (137/141)        
remote: Compressing objects: 100% (141/141), done.        

Receiving objects:   0% (1/514205)
[...]
Receiving objects:   4% (24031/514205), 16.28 MiB | 191.00 KiB/s
error: 5450 bytes of body are still expected

fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF

fatal: fetch-pack: invalid index-pack output

✗ Failed to ensure FlutterVersion(name: stable, releaseFromChannel: null, type: VersionType.channel) is cached.

Exception: Git clone failed

package:fvm/src/utils/parsers/git_clone_update_printer.dart 93  runGitCloneUpdate
package:fvm/src/services/flutter_service.dart 146               FlutterService.updateLocalMirror
package:fvm/src/workflows/ensure_cache.workflow.dart 85         ensureCacheWorkflow
package:fvm/src/commands/global_command.dart 78                 GlobalCommand.run
package:args/command_runner.dart 212                            CommandRunner.runCommand
package:fvm/src/runner.dart 246                                 FvmCommandRunner.runCommand
package:fvm/src/runner.dart 122                                 FvmCommandRunner.run
package:scope/src/scope.dart 151                                Scope.run
../runner/work/fvm/fvm/bin/main.dart 13                         main
```

**Comment by roundspecs** [comment]
@rsilvr I faced this issue on puro. This might help you.
https://github.com/pingbird/puro/issues/82

**Comment by ahmed-moha** [comment]
Why  previous versions was perfect? The issue is only these new ones

**Comment by prince-appiah** [comment]
I faced this issue about two days ago, I ran my command (fvm flutter pub get) few minutes ago and it's working now.

**Comment by bilalkhanawabzada** [comment]
try following command
fvm flutter pub get

**Comment by burhanrashid52** [comment]
The way I found to fix this is to first use HTTP/1.1 and then increase the post buffer size.

Use HTTPS/1.1
```bash
git config --global http.version HTTP/1.1
```

And then increase the buffer size using
```bash
git config --global http.postBuffer 2038500000
```

However, the trick is to increase the buffer based on the bytes mentioned in the error message that says it cannot download. For example, the git error looks like this:

```bash
    Receiving objects:   5% (28560/532810), 19.44 MiB | 336.00 KiB/s
    error: RPC failed; curl 18 transfer closed with outstanding read data remaining
    error: 5319 bytes of body are still expected
    fetch-pack: unexpected disconnect while reading sideband packet
    fatal: early EOF
    fatal: fetch-pack: invalid index-pack output
```

Here, the line `error: 5319 bytes of body are still expected` mentions 5319 bytes. So, keep increasing the buffer size until it gets small enough, i.e., zero, to get this working.
```bash
git config --global http.postBuffer {additionalbytes}
```

**Comment by pbartyik** [comment]
Spent some time around this. A better way is to set FVM_FLUTTER_URL to an ssh path like `ssh://git@github.com/flutter/flutter.git`
